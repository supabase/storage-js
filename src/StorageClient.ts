import { StorageBucketApi, StorageFileApi } from './lib'
import { Fetch } from './lib/fetch'

export class StorageClient extends StorageBucketApi {
  constructor(url: string, headers: { [key: string]: string } = {}, fetch?: Fetch) {
    super(url, headers, fetch)
  }

  /**
   * Perform file operation in a bucket.
   *
   * @param id The bucket id to operate on.
   */
  from(id: string): StorageFileApi {
    return new StorageFileApi(this.url, this.headers, id, this.fetch)
  }
}
