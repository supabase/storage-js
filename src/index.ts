export {
  StorageClient as StorageClient,
  StorageClient as SupabaseStorageClient,
} from './StorageClient'
export * from './lib/types'
