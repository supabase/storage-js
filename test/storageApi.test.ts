import { StorageClient } from '../src/index'

// TODO: need to setup storage-api server for this test
const URL = 'http://localhost:8000/storage/v1'
const KEY =
  'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJzdXBhYmFzZSIsImlhdCI6MTYwMzk2ODgzNCwiZXhwIjoyNTUwNjUzNjM0LCJhdWQiOiIiLCJzdWIiOiIzMTdlYWRjZS02MzFhLTQ0MjktYTBiYi1mMTlhN2E1MTdiNGEiLCJSb2xlIjoicG9zdGdyZXMifQ.pZobPtp6gDcX0UbzMmG3FHSlg4m4Q-22tKtGWalOrNo'

const storage = new StorageClient(URL, { Authorization: `Bearer ${KEY}` })
const newBucketName = `my-new-bucket-${Date.now()}`

describe('bucket api', () => {
  test('Build to succeed', async () => {
    // Basic test to ensure TS build is working.
    expect(true).toEqual(true)
  })

  test('Get all buckets', async () => {
    const res = await storage.listBuckets()
    expect(res.data).not.toBeNull()
  })

  test('Get bucket by id', async () => {
    const res = await storage.getBucket('bucket2')
    expect(res.data).toMatchSnapshot()
  })

  test('Get bucket with wrong id', async () => {
    const res = await storage.getBucket('not-exist-id')
    expect(res.error).toMatchSnapshot()
  })

  test('create new bucket', async () => {
    const res = await storage.createBucket(newBucketName)
    expect(res.data?.name).toEqual(newBucketName)
  })

  test('create new public bucket', async () => {
    const newPublicBucketName = 'my-new-public-bucket'
    await storage.createBucket(newPublicBucketName, { public: true })
    const res = await storage.getBucket(newPublicBucketName)
    expect(res.data!.public).toBe(true)
  })

  test('update bucket', async () => {
    const updateRes = await storage.updateBucket(newBucketName, { public: true })
    expect(updateRes.error).toBeNull()
    expect(updateRes.data).toMatchSnapshot()
    const getRes = await storage.getBucket(newBucketName)
    expect(getRes.data!.public).toBe(true)
  })

  test('empty bucket', async () => {
    const res = await storage.emptyBucket(newBucketName)
    expect(res.error).toBeNull()
    expect(res.data).toMatchSnapshot()
  })

  test('delete bucket', async () => {
    const res = await storage.deleteBucket(newBucketName)
    expect(res.error).toBeNull()
    expect(res.data).toMatchSnapshot()
  })
})
