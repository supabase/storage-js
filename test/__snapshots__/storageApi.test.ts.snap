// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bucket api Get bucket by id 1`] = `
Object {
  "created_at": "2021-02-17T04:43:32.770206+00:00",
  "id": "bucket2",
  "name": "bucket2",
  "owner": "4d56e902-f0a0-4662-8448-a4d9e643c142",
  "public": false,
  "updated_at": "2021-02-17T04:43:32.770206+00:00",
}
`;

exports[`bucket api Get bucket with wrong id 1`] = `[StorageApiError: The resource was not found]`;

exports[`bucket api delete bucket 1`] = `
Object {
  "message": "Successfully deleted",
}
`;

exports[`bucket api empty bucket 1`] = `
Object {
  "message": "Successfully emptied",
}
`;

exports[`bucket api update bucket 1`] = `
Object {
  "message": "Successfully updated",
}
`;
